### PR TITLE
[Datasets] Split `test_dataset_formats` into separate modules

### LINKED ISSUE #28398: [Datasets] Split `test_dataset_formats` into separate modules
### Description

Split `test_dataset_formats` into modules like `test_json`, `test_csv`, and `test_parquet`.

### Use case

`test_dataset_formats` is 3282 lines long. Splitting `test_dataset_formats` would make the tests easier to read and maintain.

### PR DESCRIPTION
**Pull request by PaulFenton**
## Why are these changes needed?

Solves the "good first issue" task suggested by @bveeramani: https://github.com/ray-project/ray/issues/28398#issuecomment-1251762583

**Note:** This PR is primarily just moving tests into more specific files without changing them. The only exception is a workaround that was added to fix an issue with the test `test_image_folder_reader_estimate_data_size` by adding `shutdown_only` fixture to the previously run test (`test_image_folder_datasource_e2e`)

## Related issue number
Closes #28398 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by bveeramani — COMMENTED** [review]
Thanks for opening this PR! I skimmed it and it looks reasonable to me. 

Would it make sense to move the binary and text tests to separates modules?

Will perform a more thorough review later this week.

**Comment by PaulFenton** [comment]
@bveeramani I believe I have just resolved the "binary and text tests to separate modules" suggestion with the most recent commits.

File: python/ray/data/tests/test_dataset_formats.py [hunk 1]
```diff
@@ -1,58 +1,26 @@
 import os
-import shutil
```
Comment (bveeramani) [inline python/ray/data/tests/test_dataset_formats.py:2]: There's a recently-added `test_read_tfrecords`. Could you merge master in and check that it's here? Want to make sure no tests go missing.

**Review by bveeramani — CHANGES_REQUESTED** [review]
Sorry for the delayed review, and thanks again! This'll be helpful.

LGTM except couple minor things:
* I think `test_read_text_remote_args` might be in the wrong module.
* Could you merge master in and make sure `test_read_tfrecords` is here?

**Comment by bveeramani** [comment]
@clarkzinzow @c21 could you take a look and let me know if this is okay with you? I checked and all of the tests on master are there except for `test_read_tfrecords`.

**Review by clarkzinzow — APPROVED** [review]
Thank you for doing this @PaulFenton, this has been a long-standing TODO that will greatly improve the Datasets developer experience!

**Review by c21 — APPROVED** [review]
Thanks for the work!

**Review by bveeramani — APPROVED** [review]
Sweet! LGTM 🚀

**Comment by clarkzinzow** [comment]
Thanks again @PaulFenton for the contribution! 🎉

**Pull request merged**